packages/react-components/src/components/Search/Search.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import cx from 'clsx';
+import {
+  Search as SearchIcon,
+  Close,
+} from '@livechat/design-system-icons/react/material';
+import { Icon, IconTypeName } from '../Icon';
+import { Loader } from '../Loader';
+
+import styles from './Search.module.scss';
+import { KeyCodes } from '../../utils/keyCodes';
+
+const baseClass = 'search-input';
+const inputBaseClass = `${baseClass}__input`;
+
+export const enum SearchSize {
+  Compact = 'compact',
+  Medium = 'medium',
+  Large = 'large',
+}
+
+export interface ISearchProps {
+  isControlledSubmit?: boolean;
```

Comment:
I don't understand the purpose of this flag. I know what controlled, and uncontrolled inputs are but in this case, using these flags changes this component from calling `onChange` on actual input change to on entering key down. Such behavior is really weird and unintuitive from the consumer's point of view. If we really want to give such a possibility to additionally handle "submitting," then this should have its own callback `onSubmit` and not reusing `onChange`. On the other hand, having an input that can only be submitted from the keyboard without any action button to do the same thing is also weird and not clear for the end-user.

IMO I would remove this prop and logic related to it. In order to have controlled / uncontrolled versions of this component, we just need to respect potential changes in `value` prop during the component lifetime and override the internal `searchValue` by it. It can be achieved using simple `useEffect`:

```ts
React.useEffect(() => {
    if (typeof value === 'string') {
      setSearchValue(value);
    }
  }, [value]);
```